### PR TITLE
Add George Grigorev submission

### PR DESCRIPTION
**Pull request by thepowerfuldeez**
Naive implementation of pre-norm rotary Transformer from assignment with some tweaks:
- muP initialization and LR multipliers
- larger model and context: 512 context, 8 layers, d_model=768, d_ff=2048, n_heads=12, batch size=96, total steps=28000
- bf16 training with grad scaler
- gradient accumulation = 2
- Muon optimizer for 2d params with lr 1.725e-2, lr_min=1.15e-3, wd=1.15e-4, momentum=0.95 with 300 steps warmup 
- cosine lr schedule with warmup steps = 1000
- QK-norm
- LayerNorm-scaling (by block position)
- aux Z-loss with weight 1e-4

### CODE REVIEW

**Comment by marcelroed** [comment]
Feel free to submit as many updates as you want! Glad you've enjoyed the course.
